http-webhook.md

```diff
@@ -0,0 +1,339 @@
+
+All further error status codes apply as specified in [RFC7231][RFC7231].
+
+## 3. Authorization
+
+The delivery request MUST use one of the following two methods, both of which
+lean on the OAuth 2.0 Bearer Token [RFC6750][RFC6750] model.
+
+The delivery target MUST support both methods.
+
+The client MAY use any token-based authorization scheme. Challenge-based
+schemes MUST NOT be used.
+
+### 3.1.  Authorization Request Header Field
+
+The access token is sent in the [`Authorization`][Authorization] request
+header field defined by HTTP/1.1.
+
+For [OAuth 2.0 Bearer][Bearer] tokens, the "Bearer" scheme MUST be used.
+
+Example:
+
+``` text
+POST /resource HTTP/1.1
+Host: server.example.com
+Authorization: Bearer mF_9.B5f-4.1JqM
+```
+
+### 3.2 URI Query Parameter
+
+When sending the access token in the HTTP request URI, the client adds the
+access token to the request URI query component as defined by "Uniform Resource
+Identifier (URI): Generic Syntax" [RFC3986][RFC3986], using the "access_token"
+parameter.
+
+For example, the client makes the following HTTP request:
+
+``` text
+POST /resource?access_token=mF_9.B5f-4.1JqM HTTP/1.1
+Host: server.example.com
+```
+
+The HTTP request URI query can include other request-specific parameters, in
+which case the "access_token" parameter MUST be properly separated from the
+request-specific parameters using "&" character(s) (ASCII code 38).
+
+For example:
+
+    https://server.example.com/resource?access_token=mF_9.B5f-4.1JqM&p=q
+
+Clients using the URI Query Parameter method SHOULD also send a
+Cache-Control header containing the "no-store" option.  Server
+success (2XX status) responses to these requests SHOULD contain a
+Cache-Control header with the "private" option.
+
+Because of the security weaknesses associated with the URI method (see
+[RFC6750, Section 5][RFC6750]), including the high likelihood that the URL
+containing the access token will be logged, it SHOULD NOT be used unless it is
+impossible to transport the access token in the "Authorization" request header
+field or the HTTP request entity-body. All further caveats cited in
+[RFC6750][RFC6750] apply equivalently.
+
+## 4. Abuse Protection
+
+Any system that allows registration of and delivery of notifications to
+arbitary HTTP endpoints can potentially be abused such that someone maliciously
+or inadvertently registers the address of a system that does not expect such
+requests and for which the registering party is not authorized to perform such
+a registration. In extreme cases, a notification infrastructure could be abused
+to launch denial-of-service attacks against an arbitrary web-site.
+
+To protect the sender from being abused in such a way, a legitimate delivery
+target needs to indicate that it agrees with notifications being delivered to it.
+
+Reaching the delivery agreement is realized using the following validation
+handshake. The handshake can either be executed immediately at registration
```

Comment:
nit: the language "either... or" is a false dichotomy. E.g. a K8S style API might call this endpoint both at resource creation _and_ periodically out of band.

Reply:
What's the requirement for doing the handshake twice?